docs/servicedns-with-externaldns.md

```diff
@@ -0,0 +1,250 @@
+```bash
+$ gcloud dns managed-zones create "external-example-com" \
+    --dns-name "external.example.com." \
+    --description "Example zone for ExternalDNS"
+```
+
+Make a note of the nameservers that were assigned to your new zone.
+```bash
+$ gcloud dns record-sets list \
+    --zone "external-example-com" \
+    --name "external.example.com." \
+    --type NS
+NAME                   TYPE  TTL    DATA
+external.example.com.  NS    21600  ns-cloud-b1.googledomains.com.,ns-cloud-b2.googledomains.com.,ns-cloud-b3.googledomains.com.,ns-cloud-b4.googledomains.com.
+```
+
+Tell the parent zone where to find the DNS records for `external.example.com` by adding the corresponding NS records
+in the parent zone. The following example uses the parent zone `example-com` and domain `example.com`.
+```bash
+$ gcloud dns record-sets transaction start --zone "example-com"
+$ gcloud dns record-sets transaction add ns-cloud-b{1..4}.googledomains.com. \
+    --name "external.example.com." --ttl 300 --type NS --zone "example-com"
+$ gcloud dns record-sets transaction execute --zone "example-com"
+```
+
+Deploy the ExternalDNS controller. The `domain-filter` argument is used to filter hosted zones matching
+domain name `external.example.com`. Omit this flag to process all available hosted zones.
+```bash
+$ cat <<EOF | kubectl create -f -
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: external-dns
+  namespace: federation-system
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:v0.5.7
+        args:
+        - --source=crd
+        - --crd-source-apiversion=multiclusterdns.federation.k8s.io/v1alpha1
+        - --crd-source-kind=DNSEndpoint
+        - --provider=google
+        - --domain-filter=external.example.com
+        - --registry=txt
+        - --txt-owner-id=my-identifier
+EOF
+```
+
+Deploy test `FederatedDeployment` and `FederatedService` resources:
+```bash
+$ cat <<EOF | kubectl create -f -
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedDeployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      replicas: 2
+      selector:
+        matchLabels:
+          app: nginx
+      template:
+        metadata:
+          labels:
+            app: nginx
+        spec:
+          containers:
+          - image: nginx
+            name: nginx
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedDeploymentPlacement
+metadata:
+  name: test-deployment
+spec:
+  clusterNames:
+  - cluster1
+  - cluster2
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedService
+metadata:
+  name: test-service
+spec:
+  template:
+    spec:
+      selector:
+        app: nginx
+      type: LoadBalancer
```

Comment:
question: we only support `LoadBalancer` type service here?

Reply:
Yes. I created https://github.com/kubernetes-sigs/federation-v2/issues/370 to add `NodePort` support. Please comment on the issue if you would like to see other types supported.